File: Packages/leetcode/LengthOfLongestSubstring.java
class Solution {
	public int lengthOfLongestSubstring(String s) {
		int length = 0;
		String substring = "";
		for (int i = 0; i < s.length(); i++) {
			char c = s.charAt(i);
			substring = substring.substring(substring.indexOf(c)+1) + c;
			length = Math.max(length, substring.length());
		}
		return length;
	}
}

File: Projects/leetcode/lengthOfLongestSubstring.kt
internal class Solution {
    fun lengthOfLongestSubstring(s: String): Int {
        var length = 0
        var substring = ""
        for (i in 0 until s.length()) {
            val c = s.charAt(i)
            substring = substring.substring(substring.indexOf(c) + 1) + c
            length = Math.max(length, substring.length())
        }
        return length
    }
}
